### PR TITLE
feat: lowercase all keywords

### PR DESCRIPTION
**Pull request by evilebottnawi**
:star: 

Other improvement:
1. Support alternative syntax
2. NULL => `null`
3. magic constant to upper case `__line__` => `__LINE__`

Parser problem:
1. https://github.com/glayzzle/php-parser/issues/117

### CODE REVIEW

File: src/printer.js [hunk 1]
```diff
@@ -1053,6 +1053,8 @@ function printNode(path, options, print) {
         return "array";
       } else if (node.name === "\\callable") {
         return "callable";
+      } else if (node.name.toLowerCase() === "null") {
```
Comment (evilebottnawi) [inline src/printer.js]: By PSR-2

File: src/printer.js [hunk 2]
```diff
@@ -166,14 +167,25 @@ function printExpression(path, options, print) {
...
+      case "bin": {
+        let type = node.type;
+
+        if (type.toLowerCase() === "and") {
+          type = "&&";
+        }
+
+        if (type.toLowerCase() === "or") {
+          type = "||";
+        }
```
Comment (evilebottnawi) [inline src/printer.js]: More pretty

Comment (czosel) [reply src/printer.js]: Does this also handle the difference between `||` and `or`? (https://stackoverflow.com/questions/17710924/logical-operators-or-vs-double-pipe-in-php)

File: tests/operations/__snapshots__/jsfmt.spec.js.snap [hunk 3]
```diff
@@ -100,8 +100,8 @@ $a <= $b;
 $a >= $b;
 $a <=> $b;
 
-$a and $b;
-$a or $b;
+$a && $b;
+$a || $b;
```
Comment (czosel) [inline tests/operations/__snapshots__/jsfmt.spec.js.snap]: `or` and `||` are not the same - we can't make this substitution.

File: src/printer.js (hunk 2 above)
Comment (czosel) [reply src/printer.js]: :+1:

**Comment by evilebottnawi** [comment]
Seems we don't account short from :confused: Let's do this

File: tests/switch/__snapshots__/jsfmt.spec.js.snap [hunk 4]
```diff
@@ -21,6 +25,10 @@ switch (2) {
   case 2:
     $test = 'second';
     break;
+  case 10:
+  case 20:
```
Comment (evilebottnawi) [inline tests/switch/__snapshots__/jsfmt.spec.js.snap]: Pretty :+1:

File: tests/case/__snapshots__/jsfmt.spec.js.snap [hunk 5]
```diff
@@ -0,0 +1,457 @@
...
+  {
+    echo self::CONSTANT;
+  }
+
+  protected function MyProtected()
+  {
+    require ('somefile.php');
+    require_once ('somefile.php');
+    include ('somefile.php');
+    include_once ('somefile.php');
```
Comment (evilebottnawi) [inline tests/case/__snapshots__/jsfmt.spec.js.snap]: `require ('somefile.php');` => `require 'somefile.php';` 

All popular framework use this form
https://github.com/laravel/framework/search?l=PHP&q=require&type=&utf8=%E2%9C%93
https://github.com/symfony/symfony/search?utf8=%E2%9C%93&q=require&type=
https://github.com/yiisoft/yii2/search?l=PHP&q=require&type=&utf8=%E2%9C%93

Comment (evilebottnawi) [reply tests/case/__snapshots__/jsfmt.spec.js.snap]: Let's do this in other PR better :smile:

File: tests/exit/__snapshots__/jsfmt.spec.js.snap [hunk 6]
```diff
@@ -4,14 +4,13 @@ exports[`exit.php 1`] = `
 <?php
 exit("exit info");
 exit(1);
-__halt_compiler();
-echo "this is not parsed anymore!";
+die("die info");
```
Comment (evilebottnawi) [inline tests/exit/__snapshots__/jsfmt.spec.js.snap]: Move test to `half` directory

File: tests/exit/exit.php [hunk 7]
```diff
@@ -1,5 +1,5 @@
 <?php
 exit("exit info");
 exit(1);
-__halt_compiler();
-echo "this is not parsed anymore!";
+die("die info");
```
Comment (evilebottnawi) [inline tests/exit/exit.php]: Already reported in php parser

File: tests/if/__snapshots__/jsfmt.spec.js.snap [hunk 8]
```diff
@@ -24,6 +26,8 @@ if (true) {
   return 1;
 } elseif (false) {
   return 2;
+} elseif (null) {
```
Comment (evilebottnawi) [inline tests/if/__snapshots__/jsfmt.spec.js.snap]: :+1:

**Comment by evilebottnawi** [comment]
/cc @czosel @mgrip ready for review

In next PR i will fix handling mixed content:
```php
<div>Test</div>
<?php echo 'test'; ?>
<div>Foo</div>

**Comment by czosel** [comment]
Puh, this one got quite big - would you mind splitting out the parts that have nothing to do with case sensitivity into separate PRs?

**Comment by czosel** [comment]
Concerning mixed content: I think proper support for PHP in HTML is still blocked by proper HTML support in prettier, right? (See https://github.com/prettier/prettier/pull/3521)

**Comment by evilebottnawi** [comment]
@czosel no, we break code with mixed content right now, also not all mixed content is HTML, we should print mixed content as is to allow don't break code.

**Comment by evilebottnawi** [comment]
@czosel it is will difficult separate this PR on part some parts depend on other part. We can merge this as is and add this to changelog. Also right now we don't release any version and can merge big PR after review, it is normal. We will do same in prettier between release.

**Comment by evilebottnawi** [comment]
@mgrip what do you think about above?

**Comment by mgrip** [comment]
Sorry haven't had a chance to look through yet but I will today!

File: src/printer.js (hunk 1 above)
Comment (mgrip) [reply src/printer.js]: can you add a comment explaining we prefer lowercase?

File: src/printer.js [hunk 9]
```diff
@@ -1030,9 +1050,9 @@ function printStatement(path, options, print) {
     case "silent":
       return concat(["@", path.call(print, "expr")]);
     case "halt":
-      return concat(["__halt_compiler();", node.after]);
-    //@TODO: leaving eval until we figure out encapsed https://github.com/prettier/prettier-php/pull/2
+      return concat(["__halt_compiler();", node.after.replace(/\n$/, "")]);
```
Comment (mgrip) [inline src/printer.js]: add comment explaining what the `.replace()` is for

File: src/clean.js [hunk 10]
```diff
@@ -14,6 +14,14 @@ function clean(ast) {
         : { kind: ast.kind, level };
     }
   }
+
+  if (ast.kind === "magic") {
```
Comment (mgrip) [inline src/clean.js]: can you add a comment for both of these explaining we change case in the printer?

**Comment by evilebottnawi** [comment]
@mgrip i think comments about lowercase a bit overkill, by default we always prefer lowercase

**Comment by evilebottnawi** [comment]
@czosel @mgrip after merge separate PR i rebase this PR and we can merge :+1:

**Comment by mgrip** [comment]
@evilebottnawi thanks for separating everything! I think at this point everything else got merged so probably good to rebase now! Might even be easier to throw this out and start a new MR for the case stuff, up to you

**Comment by evilebottnawi** [comment]
@mgrip need review https://github.com/prettier/plugin-php/pull/92, also i can do other PR to tests `else if` and need also review

**Pull request merged**